### PR TITLE
[15.0][IMP] hr_attendance_geolocation: Change js to prevent full override attendance_manual() method.

### PR DESCRIPTION
**Pull request by victoralmau**
Change js to prevent full override `attendance_manual()` method.

TODO: The code should be improved to avoid having a private method (_manual_attendance) instead of calling the super (update_attendance) and to allow compatibility with other modules.

Please @chienandalu can you review it?

@Tecnativa

### CODE REVIEW

File: hr_attendance_geolocation/static/src/js/attendance_geolocation.js [hunk 1]
```diff
@@ -29,15 +29,18 @@ odoo.define("hr_attendance_geolocation.attendances_geolocation", function (requi
         },
         _manual_attendance: function (position) {
             var self = this;
+            var ctx = _.extend({}, session.user_context, {
+                latitude: position.coords.latitude,
+                longitude: position.coords.longitude,
+            });
```
Comment (chienandalu) [inline hr_attendance_geolocation/static/src/js/attendance_geolocation.js]:
```suggestion
            var const = Object.assign(session.user_context, {
                latitude: position.coords.latitude,
                longitude: position.coords.longitude,
            });
```

File: hr_attendance_geolocation/static/src/js/attendance_geolocation.js [hunk 2]
```diff
@@ -106,16 +109,15 @@ odoo.define("hr_attendance_geolocation.attendances_geolocation", function (requi
                 );
                 pinBoxVal = this.$(".o_hr_attendance_PINbox").val();
             }
+            var ctx = _.extend({}, session.user_context, {
+                latitude: position.coords.latitude,
+                longitude: position.coords.longitude,
+            });
```
Comment (chienandalu) [inline hr_attendance_geolocation/static/src/js/attendance_geolocation.js]: Better to go with vanilla JS and avoid underscore as much as possible
```suggestion
            const ctx = Object.assign(session.user_context, {
                latitude: position.coords.latitude,
                longitude: position.coords.longitude,
            });
```

File: hr_attendance_geolocation/tests/test_hr_attendance_geolocation.py [hunk 3]
```diff
@@ -14,9 +15,9 @@ def setUp(self):
 
     def test_attendance_geolocation(self):
         # Called from js
-        self.employee.attendance_manual(
-            "hr_attendance.hr_attendance_action_my_attendances", None, self.location
-        )
+        self.employee.with_context(
+            latitude=self.location[0], longitude=self.location[1]
+        ).attendance_manual({})
```
Comment (chienandalu) [inline hr_attendance_geolocation/tests/test_hr_attendance_geolocation.py]:
```suggestion
        lat, long = self.location
        self.employee.with_context(
            latitude=lat, longitude=long
        ).attendance_manual({})
```

**Review by chienandalu — APPROVED** [review]
I tried to test it on runbot but it seems that the browser considers it non secure so it won't gather the geolocation.

Non blocking comments:

**Comment by victoralmau** [comment]
> I tried to test it on runbot but it seems that the browser considers it non secure so it won't gather the geolocation.

You are right, it cannot currently be tested on runboat because https is not available. (related to https://github.com/OCA/hr-attendance/pull/80#issuecomment-1308693121)

> Non blocking comments:

Thanks, changes done.

**Comment by chienandalu** [comment]
Ok, then :)

**Comment by pedrobaeza** [comment]
Just change `http` to `https` on the URL and you will be able to test it accepting the warning.

**Comment by chienandalu** [comment]
Ok, thanks. Tested then :smile:

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**